### ISSUE TITLE
add back a true black (amoled/battery-friendly) theme

### ISSUE THREAD

**Issue by konradmoesch** [mollyim/mollyim-android#309]
### Is there an existing request for this?

- [X] I have searched the existing issues

### Feature description

I'd like to request adding back a true black theme. I feel that there is a need for it (in #128, most of the reactions to the comments seem to be unsatisfied that this was simply removed, without properly supporting both dark themes) and personally, most of my apps do support it.
Please add an option for a true dark background back! I think I am not alone when I say that this was one of the reasons for me to switch to molly, and for now I will have to stay on pre-v2.7.2-1...

**Comment by konradmoesch** [comment]
Adding an AMOLED theme should be pretty straightforward, I have done this for my apps via a theme overlay.
The implementation in https://github.com/jarnedemeulemeester/findroid is very similar and Molly could do something similar:

in [`values-night.xml`](https://github.com/jarnedemeulemeester/findroid/blob/06f5380d0db3013d1c12134a77b43b85fd33ed28/core/src/main/res/values-night/themes.xml#L19), the background color gets overriden to black and all suface colors are also set to reference this color:
```
<style name="ThemeOverlay.Molly.Amoled" parent="">
        <!-- Surface -->
        <item name="android:colorBackground">@android:color/black</item>
        <item name="colorSurface">@android:color/black</item>
        <item name="colorSurfaceContainer">?attr/colorSurface</item>
        <item name="colorSurfaceContainerLow">?attr/colorSurface</item>
        <item name="colorSurfaceContainerHigh">?attr/colorSurface</item>
</style>
```
Then, one only has to add a preference, and the ThemeOverlay has to be applied, if enabled:
in [MainActivity](https://github.com/jarnedemeulemeester/findroid/blob/06f5380d0db3013d1c12134a77b43b85fd33ed28/app/phone/src/main/java/dev/jdtech/jellyfin/MainActivity.kt#L152)
```
setTheme(CoreR.style.ThemeOverlay_Findroid_Amoled)
```

Unfortunately, I was unable to implement this myself (there are so many different (for me quite confusingly named) `[..]colors.xml` that just link to each other and there seems to be at least two different preference systems (`TextSecurePreferences` and `SignalStore`, with no documentation/comments whatsoever I could find), so I was unable to really grok your code.

With the above code snippets, however, it should be very straightforward for you to implement a ThemeOverlay and adapt it to Molly, I think. This would mean very low maintenance overhead, too.
Really looking forward to seeing this implemented:)

**Comment by valldrac** [comment]
Hey! Thanks for the sharing. I've thought about this approach before, but it got sidelined for other priorities after discussing it with the community.

I've got some code snippets for it actually. Feel free to use them as a starting point. Looking forward to seeing your pull-request!

https://gist.github.com/valldrac/6867c8fad3b923a32b3bc06ddca92ccc

**Comment by thekyber** [comment]
I have been using a custom fork of Signal for more then a year now that I change the main dark color to black using GitHub Actions, maybe you can do it also and release a separate variant that has the AMOLED theme, instead of maintaining an extra theme at one app.

You can take a look here:
https://github.com/thekyber/Signal-AMOLED/blob/56931050ed7c60a5465676c176b9cd5f54fba5d8/.github/workflows/check-build-release.yml#L110

**Comment by CyberScooby** [comment]
This theme is more than just for looks. It also provides important benefits, such as:

- Reduced battery usage (lets face it, our screens are big battery hogs, so the darker the better - true black uses zero power)
- Reduced eye strain (many of us are in front of a screen all day for work and then outside of work as well)

**Comment by j12i** [comment]
There are also chronically ill people who are super sensitive to light and some of them can't use an app at all most of the time unless it has a pure black theme.

**Comment by chexo3** [comment]
Yes, I was just speaking with someone who needs a pure black theme to be able to use Signal. If it helps I could probably try and organize a bounty I have a bit of money I could contribute to that.
